backend/corpora/common/upload_sfn.py

```diff
@@ -25,3 +38,55 @@ def start_upload_sfn(collection_uuid, dataset_uuid, url):
         input=json.dumps(input_parameters),
     )
     return response
+
+
+def upload(
```

Comment:
This function factors out the non-HTTP part of the original `upload_from_link()` function so that it can be used as a common path between the portal and curation APIs.